src/goto-programs/goto_convert.cpp

```diff
@@ -347,11 +347,46 @@ void goto_convertt::convert_label(
   // magic thread creation label?
   if(has_prefix(id2string(label), "__CPROVER_ASYNC_"))
   {
-    // this is like a START_THREAD
-    codet tmp_code(ID_start_thread);
-    tmp_code.copy_to_operands(code.op0());
-    tmp_code.add_source_location()=code.source_location();
-    convert(tmp_code, tmp);
+    // A: START_THREAD : C
+    // B: GOTO Z
+    // C  SKIP
+    // D: BLOCK
+    // E: END_THREAD
+    // Z: SKIP
+
+    thread_started_counter+=1;
+    const std::string &lbl1 = "TS_" +
+      std::to_string(thread_started_counter);
+    thread_started_counter+=1;
+    const std::string &lbl2 = "TS_" +
+      std::to_string(thread_started_counter);
```

Comment:
Introducing labels here caters the risk of clashing with labels that the original source code uses. At least use a __CPROVER prefix.

Reply:
Also might be worth pulling the `TS` prefix into an `ID_ThreadStartLabel` so easier to identify and fix and clashes.